src/parser/paladin/protection/modules/talents/FirstAvenger.tsx

```diff
@@ -0,0 +1,106 @@
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent, DamageEvent } from 'parser/core/Events';
+import SPELLS from 'common/SPELLS';
+import React from 'react';
+import Statistic from 'interface/statistics/Statistic';
+import STATISTIC_ORDER from 'interface/others/STATISTIC_ORDER';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import { formatNumber } from 'common/format';
+import SpellLink from 'common/SpellLink';
+import BoringSpellValue from 'interface/statistics/components/BoringSpellValue';
+
+// One second bounce buffer for AS to bounce around and hit targets.
+const AVENGERS_SHIELD_BOUNCE_BUFFER: number = 1000;
+const TALENTLESS_NUM_OF_BOUNCES = 3;
+
+class FirstAvenger extends Analyzer {
+  lastAvengersShieldCastTimestamp: number = 0;
+  totalNumHits: number = 0;
+  totalNumCasts: number = 0;
+  castToHitsMap: Map<number, DamageEvent[]> = new Map<number, DamageEvent[]>();
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.FIRST_AVENGER_TALENT.id);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.AVENGERS_SHIELD), this.trackAvengersShieldCasts);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.AVENGERS_SHIELD), this.trackAvengersShieldHits);
+  }
+
+  trackAvengersShieldCasts(event: CastEvent): void {
+    this.lastAvengersShieldCastTimestamp = event.timestamp;
+    this.totalNumCasts += 1;
+  }
+
+  trackAvengersShieldHits(event: DamageEvent): void {
+    if (event.timestamp - this.lastAvengersShieldCastTimestamp < AVENGERS_SHIELD_BOUNCE_BUFFER) {
+      if (!this.castToHitsMap.has(this.lastAvengersShieldCastTimestamp)) {
+        this.castToHitsMap.set(this.lastAvengersShieldCastTimestamp, []);
+      }
+      this.totalNumHits += 1;
+      const hits: DamageEvent[] | undefined = this.castToHitsMap.get(this.lastAvengersShieldCastTimestamp);
```

Comment:
This *shouldn't* do anything, some kind of leftover from a previous version?

Reply:
I'm a bit shaky on how Javascript handles array references (I'm a backend Java dev myself), but this roundabout way of pushing a `DamageEvent` into the map value is succeeding according to the debug logs I put in place (adding a `console.dir` to the `statistic()` call). The `Map` values are then used in calculating the extra damage from `Avenger's Shield` that can be attributed to `First Avenger`.

![image](https://user-images.githubusercontent.com/6363390/97313901-5e01d600-1824-11eb-9bca-0b48c5ff3801.png)

Reply:
You're right, my bad. Not a primitive, thats why its passed by reference.
Said nothing